ext/mbstring/tests/sjis_encoding.phpt

```diff
@@ -0,0 +1,61 @@
+--TEST--
+Exhaustive test of Shift-JIS encoding verification and conversion
+--SKIPIF--
+<?php
+extension_loaded('mbstring') or die('skip mbstring not available');
+if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+?>
+--FILE--
+<?php
+srand(999); /* Make results consistent */
+include('encoding_tests.inc');
+mb_substitute_character(0x25); // '%'
+
+/* Read in the table of all characters in Shift-JIS */
+readConversionTable(__DIR__ . '/data/SHIFTJIS.txt', $validChars, $fromUnicode);
+
+for ($i = 0; $i < 0x20; $i++) {
+  $validChars[chr($i)] = "\x00" . chr($i);
+  $fromUnicode["\x00" . chr($i)] = chr($i);
+}
+
```

Comment:
Maybe point to https://www.unicode.org/Public/MAPPINGS/OBSOLETE/EASTASIA/ReadMe.txt here, which lists at least some of those as possible mapping errors.

Reply:
Thanks, will look at it.